applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_gradient_projection.py

```diff
@@ -199,27 +199,37 @@ def __computeControlPointUpdate(self):
             solver,
             s,
             c)
-        s_plus_c = KM.Vector(s+c)
-        eps = 1e-12
+        eps = 1e-16
 
-        if c.norm_inf() != 0.0:
+        if c.norm_inf() > eps:
             if c.norm_inf() <= self.max_correction_share * self.step_size:
                 delta = self.step_size - c.norm_inf()
                 s_inf_norm = s.norm_inf()
                 if s_inf_norm > eps :
-                    s *= delta/s.norm_inf()
+                    s *= delta/s_inf_norm
+                else:
```

Comment:
the else conditions in this commit are not really doing anything, since multiplying a vector with a max norm of (numerically) 0 wit something will still result in a (numerically) 0 vector.

Reply:
I know. I want to be sure. I have a case which was complaining NaNs so I put them there .. !

Reply:
Now these unnecessary else statements are removed !